File: lab03/AimsProject/src/hust/soict/itep/test/cart/CartTest.java
package hust.soict.itep.test.cart;

import hust.soict.itep.aims.cart.Cart_HoTuanAnh20226100;
import hust.soict.itep.aims.disc.DigitalVideoDisc_HoTuanAnh20226100;

public class CartTest {
    public static void main(String[] args) {
        Cart_HoTuanAnh20226100 cart = new Cart_HoTuanAnh20226100();
        DigitalVideoDisc_HoTuanAnh20226100 dvd1 = new DigitalVideoDisc_HoTuanAnh20226100("The Lion King","Animation",
                "Roger Allers",87,19.95f);
        cart.addDigitalVideoDisc(dvd1);
        DigitalVideoDisc_HoTuanAnh20226100 dvd2 = new DigitalVideoDisc_HoTuanAnh20226100("Star wars","Science Fiction",
                "Geogre Lucas",87,24.95f);
        cart.addDigitalVideoDisc(dvd2);
        DigitalVideoDisc_HoTuanAnh20226100 dvd3 = new DigitalVideoDisc_HoTuanAnh20226100("Aladin","Animation",18.99f);
        cart.addDigitalVideoDisc(dvd3);

        cart.print();

        cart.searchById(3);
        cart.searchById(4);

        cart.searchByTitle("The Lion King");
        cart.searchByTitle("abc");
    }
}

File: lab03/test/hust/soict/itep/aims/cart/CartTest.kt
package hust.soict.itep.aims.cart

import hust.soict.itep.aims.disc.DigitalVideoDisc
import org.junit.Test
import org.junit.jupiter.api.Assertions.*

class CartTest {

    private val cart = Cart()

    @Test
    fun addDvd() {
        val dvd1 = DigitalVideoDisc(1, "The Lion King", "Animation", "Roger Allers", 87, 19.95f)
        val dvd2 = DigitalVideoDisc(2, "Star wars", "Science Fiction", "Geogre Lucas", 87, 24.95f)
        cart.addDvd(dvd1, dvd2)
        assertEquals(2, cart.itemOrdered.size)
    }

    @Test
    fun searchByTitle() {
        val dvd1 = hust.soict.itep.aims.disc.DigitalVideoDisc(1, "The Lion King", "Animation", "Roger Allers", 87, 19.95f)
        val dvd2 = hust.soict.itep.aims.disc.DigitalVideoDisc(2, "Star wars", "Science Fiction", "Geogre Lucas", 87, 24.95f)
        cart.addDvd(dvd1, dvd2)
        assertEquals(dvd1, cart.searchByTitle("The Lion King"))
    }
}